src/hotspot/share/classfile/classFileParser.cpp

```diff
@@ -4314,14 +4314,7 @@ static void check_super_class_access(const InstanceKlass* this_klass, TRAPS) {
     }
 
     if (super_ik->is_sealed() && !super_ik->has_as_permitted_subclass(this_klass)) {
-      ResourceMark rm(THREAD);
-      Exceptions::fthrow(
-        THREAD_AND_LOCATION,
-        vmSymbols::java_lang_IncompatibleClassChangeError(),
-        "class %s cannot inherit from sealed class %s",
-        this_klass->external_name(),
-        super_ik->external_name());
-      return;
+      classfile_icce_error("class %s cannot inherit from sealed class %s", super_ik, CHECK);
```

Comment:
You don't want CHECK there - you know there is an exception pending. Just pass THREAD and return afterwards. This goes for all the new callsites.